### PR TITLE
Merge `v0.31.1-rc0` branch into master

### PR DESCRIPTION
**Pull request by eddddddy**
Post-release merge of the RC branch into master

### CODE REVIEW

File: tests/tape/test_qscript.py [hunk 1]
```diff
@@ -448,6 +448,16 @@ def test_specs_tape(self, make_script):
...
+    def test_specs_copy(self, make_script):
+        """Test that the copy method of specs retuns a SpecsDict."""
+        qs = make_script
+        copied_specs = qs.specs.copy()
+
+        assert isinstance(copied_specs, qml.tape.qscript.SpecsDict)
+        with pytest.warns(UserWarning, match="key is deprecated and will be removed"):
+            for k, v in qs.specs.items():
+                assert copied_specs[k] == v
+
```
Comment (albi3ro) [inline tests/tape/test_qscript.py:460]:
```suggestion
```

File: setup.py [hunk 2]
```diff
@@ -28,10 +28,11 @@
     "toml",
     "appdirs",
     "semantic-version>=2.7",
-    "autoray>=0.3.1",
+    "autoray>=0.3.1,<=0.6.3",
```
Comment (eddddddy) [inline setup.py:31]:
```suggestion
    "autoray>=0.3.1",
```

File: requirements.txt [hunk 3]
```diff
@@ -10,7 +10,8 @@ toml~=0.10
 appdirs~=1.4
 semantic_version~=2.10
 dask[delayed]~=2022.4.1
-autoray==0.3.1
+autoray>=0.3.1,<=0.6.3
```
Comment (eddddddy) [inline requirements.txt:13]:
```suggestion
autoray==0.3.1
```

File: setup.py (hunk 2 above)
Comment (albi3ro) [inline setup.py:31]:
```suggestion
    "autoray>=0.3.1",
```

**Review by albi3ro — APPROVED** [review]
Pretty sure there's no merge conflicts. Looks good to me.

**Review by mudit2812 — APPROVED** [review]

**Review by timmysilv — APPROVED** [review]

**Pull request merged**